### PR TITLE
Makefile Cleanup

### PR DESCRIPTION
**Pull request by marians**
- Makefile cleaned up according to Java and Golang version
- Dockerfile: used significantly smaller base image `node:0.10-slim`

### CODE REVIEW

**Comment by marians** [comment]
RFR @luebken @puja108

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,15 @@
-FROM node
+FROM node:0.10-slim
```
Comment (luebken) [inline Dockerfile]: > Unless you are working in an environment where only the node image will be deployed and you have space constraints, we highly recommend using the default image of this repository.

https://github.com/docker-library/docs/tree/master/node#nodeslim

Hmmm.

Comment (puja108) [inline Dockerfile]: Yeah, not sure if this is good, as one goal for the firstapp guides should be that users can use them to build their own apps with it later.

Comment (marians) [inline Dockerfile]: Since the push to the registry can take a significant amount of time, I'd really opt for smaller images at a cost of re-usability.

Comment (puja108) [inline Dockerfile]: doesn't the registry cache layers, so that if node:0.10 has been pushed before, it gets skipped?

Comment (marians) [inline Dockerfile]: Pardon me, of course. But, this also applies to the first pull if the user has never pulled this image before. Chances are low that we find an image the users already have locally, so their experience starts with a docker pull to their local machine.

Comment (puja108) [inline Dockerfile]: Just looked a bit into the [Dockerfile](https://github.com/joyent/docker-node/blob/3e0d2b62fa8524e05ef512564613863ee7f13255/0.10/slim/Dockerfile) for this. Doesn't look like anything important for running node code is missing, it just removes a lot of apt-get stuff and buildDeps

**Comment by marians** [comment]
So I figure you are all fine with the changes :)

File: Dockerfile (hunk 1 above)
Comment (luebken) [inline Dockerfile]: Question is why they say you shouldn't use it.

Comment (marians) [inline Dockerfile]: I think the risk of doing something bad here is low. I tested the image, it works, we also use it for the blog.

Comment (luebken) [inline Dockerfile]: Im not saying it is high risk. I'm just wandering. We said we want to use official images were possible. And here it is an advise not use it. If we promote this image I would like to understand why.

Comment (luebken) [inline Dockerfile]: I want to know: https://github.com/joyent/docker-node/issues/26

**Pull request merged**